apiserver/vshn/postgres/multiwatch.go

```diff
@@ -0,0 +1,41 @@
+package postgres
+
+import "k8s.io/apimachinery/pkg/watch"
+
+// MultiWatch is wrapper of multiple source watches which implements the same methods as a normal watch.Watch
+var _ watch.Interface = &MultiWatcher{}
+
+type MultiWatcher struct {
+	watchers []watch.Interface
+}
+
+// NewEmptyMultiWatch creates an empty watch
+func NewEmptyMultiWatch() *MultiWatcher {
+	return &MultiWatcher{}
+}
+
+// AddWatcher adds a watch to this MultiWatcher
+func (m *MultiWatcher) AddWatcher(w watch.Interface) {
+	m.watchers = append(m.watchers, w)
+}
+
+// Stop stops all watches of this MultiWatch
+func (m *MultiWatcher) Stop() {
+	for _, watcher := range m.watchers {
+		watcher.Stop()
+	}
+}
+
+// ResultChan aggregates all channels from all watches of this MultiWatch
+func (m *MultiWatcher) ResultChan() <-chan watch.Event {
+	out := make(chan watch.Event)
+	for _, w := range m.watchers {
+		watcher := w
+		go func() {
```

Comment:
This go routine will never return. So we'll end up with a bunch of those stuck every time someone uses the watch command.

We probably want to implement some sort of cancel/stop mechanism here.

Reply:
The watches in `multiwatch` instance should not be stopped when calling the channels as they need to send events continuously. Only when the client stops the "watching" action then the watches underneath. `multiwatch` should be closed. I see it like that.

Reply:
As already discussed @glrf , a single routing and a select statement will not work in this case. I will leave it as is. Thanks for the suggestion.